pandas/io/stata.py

```diff
@@ -1649,6 +1649,13 @@ def _do_convert_categoricals(self, data, value_label_dict, lbllist,
                         categories.append(value_label_dict[label][category])
                     else:
                         categories.append(category)  # Partially labeled
+                if len(categories) != len(set(categories)):
```

Comment:
I think its a bit more python to:

```
try:
    cat_data.categories = categories
except ValueError:
    # show your new ValueError
```

as non-unique category setting already raises.